### PR TITLE
Bicaws 2122 create next js app

### PR DESCRIPTION
**Pull request by joe-fol**
Create basic project template 
- Initialise Next js app
- Configured Eslint
- Configured prettier
- Configure cypress
- Added pre-commit hook

### CODE REVIEW

File: .github/codeql/paths.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+paths-ignore:
+  - cypress
+  - test
+  - testFixtures
+  - src/lib/config.ts
+  - src/pages/login/reset-password.tsx
```
Comment (monotypical) [inline .github/codeql/paths.yml:6]: we don't need these paths from the user service in this repo

**Review by monotypical — APPROVED** [review]
nice work! I've left one minor non-blocking comment, other than that it looks good! 🎉

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'typescript', 'python' ]
```
Comment (monotypical) [inline .github/workflows/codeql-analysis.yml:35]: we don't have any python code in this repo yet so we can get rid of this for now, it's causing the codeql ci test to fail

File: .github/workflows/update-npm-deps.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      - name: Setup node version
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16.3.0
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Update dependencies
+        run: npm run update-deps
```
Comment (daviesjamie) [inline .github/workflows/update-npm-deps.yml:24]: This action is going to fail, as the `update-deps` npm script hasn't been added to the `package.json` in this repo!

**Review by bjpirt — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    "lint:fix": "eslint . --fix",
+    "cypress:open": "cypress open",
+    "cypress:run": "cypress run",
+    "cypress:run:file": "cypress run --spec",
+    "test:unit": "jest --testPathIgnorePatterns=\"integration\" \"e2e\" --runInBand",
+    "test:unit:update": "npm run test:unit -- -u",
+    "test:integration": "jest --testPathPattern=\"integration\" --runInBand",
+    "test:ui": "./scripts/run-test-ui.sh",
+    "install:hooks": "./scripts/install-git-hooks.sh",
+    "update-deps": "ncu -u && npm install"
```
Comment (daviesjamie) [inline package.json:19]: `ncu` is a tool provided by the [`npm-check-updates`](https://github.com/ministryofjustice/bichard7-next-user-service/blob/main/package.json#L82) package - which isn't listed as a dependency for this package. So I expect if you try running the `update-deps` command you've defined here (`npm run update-deps`), it'll fail 😅 

A cheeky `npm install --save-dev npm-check-updates` should do the trick!

**Review by daviesjamie — APPROVED** [review]

**Pull request merged**